free-3.md

```diff
@@ -4,40 +4,86 @@
+of a playlist in a program like iTunes, without needing to have all the
+sources match. Additionally, by wrapping the DSL around streaming services,
+your collection can be easily listened to on the go via typical internet
+radio streaming services.
 
 ### Why you?
 _What excites you about this idea? How did you come up with it?_
 
+This is a practical idea that my roommate has previously attempted to address
+(at least in the YouTube domain), with a tool he calls
+[Snackbar](https://www.cs.hmc.edu/~wwong/playlist). While this tool is
+exciting, I now want to make an "improved" version that also enables typical
+playlist features, streaming, and inclusion of local files. I would use this
+tool so much it isn't even funny.
 
 ### Domain
 _Describe the project's domain in five words._
 
+Listening from multiple audio sources.
 
 ### Interface (syntax)
 _How might the user interact with the language? What does programming look 
 like? Why is this the right way to interact with the problem domain?_ 
 
+In some ways, I am torn about the best way to address this question. On one
+hand, like ImageOrganizer, I am interested in designing this DSL so it is
+easily incorperated into custom GUIs. That being said, a command line music
+tool doesn't feel ideal. That being said, it is possible I could design it
+as a command line tool and integrate it into a GUI as a plugin to something
+like [Foobar2000](https://www.foobar2000.org/), arguably the most customizable
+media player around.
 
 ### Operation (semantics)
 _What might happen when a program runs? How does a program interact with the
 user? What kinds of errors might occur, and how might they be communicated to
 the user?_
 
+A program will be a wrapper around various APIs, database
+requests, and streaming services. Thus, ideally, the end result will be a
+pretty typical music player that has an atypical set of sources (i.e. not
+just files). Especially if we go for a more GUI approach, it would be important
+for errors to be communicated to the user, but not in a experience breaking
+way. For instance, if there is an intruption to the internet, the removal of
+a YouTube video, etc. the problem songs should be noted but then skipped.
 
 ### Expressiveness
 _What should be easy to do in this language? What should be possible, but
 difficult? What should be impossible or very difficult?_
 
+It should be easy to add new songs, shuffle, repeat, etc. Additionally it
+should be able to stream music via typical internet radio methods without
+much effort. Furthermore, adding a weighted shuffle with a series of
+parameters would be awesome (i.e. song more recently added are more likely
+to play, song with more upvotes are more likely to play, etc.) That being said,
+it should be challenging to do things beyond this simple command/result
+interface. The most complicated scripts will be simplified wrappers around
+database commands for queries like "finding all songs listened to within the
+last 10 days."
```

Comment:
How will upvotes work?

Also, what about things like Pandora and Spotify Radio? Should they be in play here?